File: composeApp/src/androidMain/java/org/application/shikiapp/shared/HtmlUtils.java
package org.application.shikiapp.shared;

import android.graphics.Color;

import androidx.annotation.ColorInt;
import androidx.annotation.OptIn;
import androidx.media3.common.util.UnstableApi;
import androidx.media3.common.util.Util;

final class HtmlUtils {

    private HtmlUtils() {
    }

    @OptIn(markerClass = UnstableApi.class)
    public static String toCssRgba(@ColorInt int color) {
        return Util.formatInvariant(
                "rgba(%d,%d,%d,%.3f)",
                Color.red(color), Color.green(color), Color.blue(color), Color.alpha(color) / 255.0);
    }

    public static String cssAllClassDescendantsSelector(String className) {
        return "." + className + ",." + className + " *";
    }
}

File: composeApp/src/androidMain/kotlin/org/application/shikiapp/shared/utils/ui/subtitles/HtmlUtils.kt
package org.application.shikiapp.shared.utils.ui.subtitles

import android.graphics.Color
import androidx.media3.common.util.UnstableApi
import androidx.media3.common.util.Util

object HtmlUtils {
    @UnstableApi
    fun toCssRgba(color: Int) = Util.formatInvariant(
        "rgba(%d,%d,%d,%.3f)",
        Color.red(color), Color.green(color), Color.blue(color), Color.alpha(color) / 255.0
    )

    fun cssAllClassDescendantsSelector(className: String) = ".$className,.$className *"
}